include/core/truthTable/truth_table.hpp

```diff
@@ -112,6 +133,16 @@ namespace syrec {
                 cube.pop_back();
             }
 
+            [[nodiscard]] auto equals(const std::uint64_t num) const -> bool {
+                Cube cubeEq(cube);
+                return cubeEq.toInteger() == num;
+            }
+
+            [[nodiscard]] auto equals(const std::string& str) const -> bool {
+                Cube cubeEq(cube);
+                return Cube::fromString(str) == cubeEq;
+            }
```

Comment:
avoid creating unnecessary copies.
```suggestion
            [[nodiscard]] auto equals(const std::string& str) const -> bool {
                return cube == Cube::fromString(str);
            }
```

Reply:
done 👍🏼